src/pages/suggestions/edit/[suggestion].tsx

```diff
@@ -0,0 +1,193 @@
+import {
+  updateEditSuggestion,
+  updateLastDeletedInstrumentUuid,
+  updateLastEditedUuid,
+} from "@/redux/slices/edit-suggestion.slice"
+import { setActiveArea } from "@/redux/slices/new-suggestion.slice"
+
+export const getServerSideProps: GetServerSideProps = async (context) => {
+  const supabase = createPagesServerClient(context)
+  // Check if we have a session
+  const {
+    data: { session },
+  } = await supabase.auth.getSession()
+  const { params } = context
+
+  try {
+    // Check if the suggestion exists
+    const { data } = await getSuggestion(supabase, params?.suggestion as string)
+    if (data == null) return { notFound: true }
+
+    // If the suggestion has no id
+    const author: { id: string } = data?.author as { id: string }
+    if (author?.id === undefined) return { notFound: true }
+
+    // If the user does not match with the author of the suggestion
+    if (session?.user.id !== author.id)
+      return {
+        redirect: {
+          destination: "/403",
+          permanent: false,
+        },
+      }
+
+    return { props: { suggestion: data } }
+  } catch {
+    return { notFound: true }
+  }
+}
+
+interface EditSuggestionPageProps {
+  suggestion: Suggestion
+}
+
+const EditSuggestionPage = (props: EditSuggestionPageProps) => {
+  const user = useUser()
+  const supabase = useSupabaseClient<Database>()
+  const router = useRouter()
+  const dispatch = useDispatch()
+
+  const newSuggestion = useSelector((state: AppState) => state.editSuggestion)
+  const suggestion = useSelector((state: AppState) => state.editSuggestion.suggestion)
+  const activeArea = useSelector((state: AppState) => state.editSuggestion.activeArea)
+  const deletedInstruments = useSelector(
+    (state: AppState) => state.editSuggestion.deletedInstrumentUuids
+  )
+
+  //If the user visits a new edit page, clear the previously edited from redux
+  if (props.suggestion.id !== newSuggestion.lastEditedUuid) {
+    const suggestionInstruments: NewSuggestionInstrument[] = []
+    props.suggestion.song_instruments.forEach((element) => {
+      suggestionInstruments.push({
+        id: element.id,
+        description: element.description ?? "",
+        instrument: element.instrument,
+      })
+    })
+
+    dispatch(
+      updateEditSuggestion({
+        artist: props.suggestion.artist,
+        link: props.suggestion.link,
+        motivation: props.suggestion.motivation,
+        title: props.suggestion.title,
+        instruments: suggestionInstruments,
+      })
+    )
+
+    dispatch(updateLastDeletedInstrumentUuid([]))
+    dispatch(updateLastEditedUuid(props.suggestion.id))
+  }
+
+  const saveSuggestion = (onSuccess: () => void, onError: () => void) => {
+    if (user) {
+      updateSuggestion(supabase, suggestion, newSuggestion.lastEditedUuid ?? "")
+        .then((response) => {
+          if (response.error) {
+            onError()
+            return
+          }
+
+          const suggestionId = response.data.at(0)!.id
```

Comment:
This is the same as the lastEditedUuid right?

Reply:
This guarantees that the UUID always is the one from the response of the backend.